lightning/src/ln/channelmanager.rs

```diff
@@ -4915,6 +4960,45 @@ where
 
 			{
 				let per_peer_state = self.per_peer_state.read().unwrap();
+
+				{
+					// Get pending messages to be broadcasted.
+					let mut pending_broadcast_messages = self.pending_broadcast_messages.lock().unwrap();
+
+					// If we have pending broadcast events
+					if pending_broadcast_messages.ticks_remaining != None {
+						// And it is time to broadcast
+						if pending_broadcast_messages.ticks_remaining == Some(0) {
+							// And we are connected to non-zero number of peers, we can broadcast successfully
+							if per_peer_state.len() > 0 {
+								let a_peer_state_mutex = per_peer_state.values().next().unwrap();
+								let mut a_peer_state = a_peer_state_mutex.lock().unwrap();
+
+								for broadcast_evts in pending_broadcast_messages.broadcast_message.iter() {
+									a_peer_state.pending_msg_events.push(broadcast_evts.clone());
+								}
+								// After broadcasting, we clear the event vector.
+								// And set the timer to None.
+								pending_broadcast_messages.broadcast_message.clear();
+								pending_broadcast_messages.ticks_remaining = None;
+
+								log_info!(self.logger, "Successfully broadcasted the pending channel update messages.");
+							}
+							// Else we restart the counter, waiting for next time to try rebroadcasting
+							else {
+								pending_broadcast_messages.ticks_remaining = Some(PENDING_BROADCAST_MESSAGES_TIMER_TICKS);
+							}
+						}
+						// Else we decrement the counter
+						else {
+							pending_broadcast_messages.ticks_remaining = match pending_broadcast_messages.ticks_remaining {
```

Comment:
I have simplified the code in [648a214](https://github.com/lightningdevkit/rust-lightning/commit/648a214968bd904d95939e53fbbbd698d237df74)
I have chosen a simpler approach of trying to broadcast after each timer ticks instead of broadcasting after N timer ticks.